### PR TITLE
1 initialize project

### PR DESCRIPTION
**Pull request by nor-wtag**
Modifications-

1. Removed unnecessary comments.
2. Sorted Gems in the correct order.
3. Added necessary gems.
4. Established PostgreSQL connection
5. Created a database

### CODE REVIEW

File: Gemfile
Comment (rsh-wtag) [inline Gemfile:1]: Please change your `rubocop` configuration to resolve this `Check failure`.

File: config/credentials.yml.enc
Comment (rsh-wtag) [inline config/credentials.yml.enc:1]: Please add a new line at the end of the file to omit the GitHub warning.

File: Gemfile [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+gem 'jbuilder'
+gem 'letter_opener'
+gem 'pg'
+gem 'pundit'
+gem 'rails-i18n'
+gem 'redis'
+gem 'rufus-scheduler'
+gem 'sidekiq'
+gem 'sprockets-rails'
+gem 'sqlite3', '>= 1.4'
```
Comment (rsh-wtag) [inline Gemfile:17]: Do we need this gem?

File: .dockerignore
Comment (rsh-wtag) [inline .dockerignore:1]: Do we need this file?

File: Dockerfile
Comment (rsh-wtag) [inline Dockerfile:1]: Ditto.

File: Gemfile (hunk 1 above)
Comment (rsh-wtag) [inline Gemfile:17]: Do we need this gem?

Comment (rsh-wtag) [inline Gemfile:17]: Do we need this gem?

File: config/credentials.yml.enc
Comment (nor-wtag) [reply config/credentials.yml.enc:1]: Fixed it. Thank you.

File: .dockerignore
Comment (nor-wtag) [reply .dockerignore:1]: Deleted it.

File: Gemfile (hunk 1 above)
Comment (nor-wtag) [reply Gemfile]: Removed the gem.

File: Dockerfile
Comment (nor-wtag) [reply Dockerfile:1]: Removed the file.

File: Gemfile
Comment (nor-wtag) [reply Gemfile:1]: Changed it. Thank you.

**Review by ttu-wtag — APPROVED** [review]

**Review by rsh-wtag — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: .ruby-version
+          bundler-cache: true
+
+      - name: Lint code for consistent style
+        run: bin/rubocop -f github
+
```
Comment (zma-wtag) [inline .github/workflows/ci.yml:55]: You have removed the test job from this PR. But please try to introduce a rspec running job from the pr where you have introduce the rspecs.

File: bin/setup [hunk 3]
```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env ruby
```
Comment (zma-wtag) [inline bin/setup:1]: purpose of this file?

File: config/credentials.yml.enc [hunk 4]
```diff
@@ -0,0 +1 @@
+GvtRQFoB4F+DgvEa0kIigPuRnnh0lAwxkSNU5JLeGOPSXObKMq4Qx+shFxwJ6Pjc8nzI/k299tnTZiXAHL3u89zH7Ka1kTO6vW9Dvhvw9HiNBIXekSExrSI+sehofOcOz+r4blklm9NAT0U0REgP/cYUG03tCxYRmMD/XJZG5PRcBrB0MbVoe6907rP+4KIHdFZbMVUMiRazPBtIxHI+JTEW0Utj1K+M1KXxqiZYTD0dsSCVpHLoU4P5PYMWoEdcmkHIJnqV68gYoGWxBcIg6z6eGIc1+pUSRB7WcHj9yZMPjQaykGJPWs4qUY0/xmIU0DD4BNC5vurnGo6WHDCHQ2X6NbGleIrekX6R/1l65B4dgiMo/IEncp3SB/FZa7WDfboWlsVzrc64+T8bygPePdnLuA+j--Te7bYcVrlJUmM53D--57Yl8G3CZWk9uM/pBPksaA==
```
Comment (zma-wtag) [inline config/credentials.yml.enc:1]: why we need this file?

File: config/application.rb [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+# require "rails/test_unit/railtie"
+
+# Require the gems listed in Gemfile, including any gems
+# you've limited to :test, :development, or :production.
+Bundler.require(*Rails.groups)
+
+module OnlineMarketplace
+  class Application < Rails::Application
+    # Initialize configuration defaults for originally generated Rails version.
+    config.load_defaults 7.2
```
Comment (zma-wtag) [inline config/application.rb:24]: purpose of this line?

Comment (nor-wtag) [reply config/application.rb:24]: Bhai, I used it for setting the default configuration options for my Rails application.

File: config/credentials.yml.enc (hunk 4 above)
Comment (nor-wtag) [reply config/credentials.yml.enc:1]: I am using it for maintaining the security and integrity of sensitive data in my application.

File: bin/setup (hunk 3 above)
Comment (nor-wtag) [reply bin/setup:1]: Bhai, I am using this file to have setup scripts that can automate the initialization of the application.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (nor-wtag) [reply .github/workflows/ci.yml:55]: Bhai, I have added it on the next branch. Thank you.

File: Gemfile [hunk 6]
```diff
@@ -0,0 +1,31 @@
+source 'https://rubygems.org'
+gem 'bcrypt', '~> 3.1.7'
+gem 'bootsnap', require: false
+gem 'carrierwave'
+gem 'devise'
+gem 'grape'
```
Comment (evk-wtag) [inline Gemfile:6]: Let's remove these gems from this PR, and add them in the required PR where we implement these.

**Review by zma-wtag — APPROVED** [review]
Good Work!

File: Gemfile (hunk 6 above)
Comment (nor-wtag) [reply Gemfile]: Fixed it Apu. Thank you.

**Review by evk-wtag — APPROVED** [review]

**Pull request merged**